### PR TITLE
 Week3

### PR DESCRIPTION
**Pull request by Samtabanja**
Sam, Week3

### CODE REVIEW

File: homework/src/App.js [hunk 1]
```diff
@@ -17,12 +17,25 @@ class App {
     // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
 
     const root = document.getElementById('root');
+
+    const header = Util.createAndAppend('header', root, { 'html': '<h3> HYF Repositories <h3>' });
```
Comment (remarcmij) [inline homework/src/App.js]: See the feedback that @JawharBairakdar already gave you in week 2 but somehow didn't make it in your week 3 homework.

File: homework/src/App.js [hunk 2]
```diff
@@ -39,20 +52,20 @@ class App {
       const repo = this.repos[index];
       const contributors = await repo.fetchContributors();
 
+
       const container = document.getElementById('container');
```
Comment (remarcmij) [inline homework/src/App.js]: No need for this line if you previously defined `this.container`,

File: homework/src/App.js [hunk 3]
```diff
@@ -63,7 +76,7 @@ class App {
    * @param {Error} error An Error object describing the error.
    */
   renderError(error) {
-    // Replace this comment with your code
+    Util.createAndAppend('p', container, { html: error.message, class: 'alert-error' });
```
Comment (remarcmij) [inline homework/src/App.js]: `container` is undefined here, use `this.container` instead.

File: homework/src/App.js [hunk 4]
```diff
@@ -17,12 +17,25 @@ class App {
...
+    const header = Util.createAndAppend('header', root, { 'html': '<h3> HYF Repositories <h3>' });
+    const select = Util.createAndAppend('select', header);
+    Util.createAndAppend('div', root, { id: 'container' });
+
     // ...
 
     try {
       // ...
       const repos = await Util.fetchJSON(url);
       this.repos = repos.map(repo => new Repository(repo));
```
Comment (remarcmij) [inline homework/src/App.js]: The list of repositories was supposed to be sorted by repository name.

File: homework/src/Contributor.js [hunk 5]
```diff
@@ -14,5 +14,12 @@ class Contributor {
   */
   render(contributorList) {
     // Replace this comment with your code
+
+    const div3 = Util.createAndAppend('div', contributorList, { 'class': 'container' });
```
Comment (remarcmij) [inline homework/src/Contributor.js]: The name `div3` seems arbitrary in this context. There are no other `div` variables here that would warrant adding the suffix `3`.

But in fact, this should not be a `div` but an `li`.

Note also that you do not need to enclose object property names in quotes, unless those property names include characters that are not permitted in JavaScript variable names.

File: homework/src/Contributor.js [hunk 6]
```diff
@@ -14,5 +14,12 @@ class Contributor {
...
+
+
+    Util.createAndAppend('img', div3, { 'src': this.data.avatar_url });
```
Comment (remarcmij) [inline homework/src/Contributor.js]: Use a single blank line to separate code blocks.

File: homework/src/Repository.js [hunk 7]
```diff
@@ -16,19 +16,50 @@ class Repository {
     //
     // Replace this comment with your code
     //
+
+    const table = Util.createAndAppend('table', parent);
+    const tr = Util.createAndAppend('tr', table);
+    Util.createAndAppend('td', tr, { 'html': 'Repository  : ', 'class': 'tr' });
+    Util.createAndAppend('td', tr, { 'html': this.repo.name });
```
Comment (remarcmij) [inline homework/src/Repository.js]: You have repeated code here: you are creating a `tr` with two `td` children three times. You could pply the DRY principle by creating a method, e.g. `addRow`:

```js

render(parent) {
  const table = Util.createAndAppend('table', parent);
  this.addRow(table, 'Repository :', this.repo.name);
  //...
}

addRow(table, label, value) {
  //...
}
```

File: homework/src/Repository.js [hunk 8]
```diff
@@ -16,19 +16,50 @@ class Repository {
...
+
+
+
+    const tr2 = Util.createAndAppend('tr', table);
```
Comment (remarcmij) [inline homework/src/Repository.js]: Use a single blank line to separate code blocks.

File: homework/src/Repository.js [hunk 9]
```diff
@@ -16,19 +16,50 @@ class Repository {
...
+    Util.createAndAppend('td', tr3, { 'html': this.repo.forks_count });
+
+
+    const Date1 = new Date(this.repo.updated_at).toLocaleString();
+    const tr4 = Util.createAndAppend('tr', table);
+    Util.createAndAppend('td', tr4, { 'html': 'last update : ', 'class': 'tr' });
+    Util.createAndAppend('td', tr4, { 'html': Date1 });
+
+
   }
```
Comment (remarcmij) [inline homework/src/Repository.js]: Do not add blank lines before the closing brace. They serve no purpose and unnecessarily stretch out your code vertically.

File: homework/src/App.js [hunk 10]
```diff
@@ -39,20 +52,20 @@ class App {
...
       const container = document.getElementById('container');
       // Erase previously generated inner HTML from the container div
       container.innerHTML = '';
 
-      const leftDiv = Util.createAndAppend('div', container);
-      const rightDiv = Util.createAndAppend('div', container);
-
-      const contributorList = Util.createAndAppend('ul', rightDiv);
+      const leftDiv = Util.createAndAppend('div', container, { id: 'div1' });
+      const rightDiv = Util.createAndAppend('div', container, { id: 'div2' });
```
Comment (remarcmij) [inline homework/src/App.js]: This `rightDiv` should actually be a list (`ul`). Using a semantically appropriate element helps to make your app ARIA-compliant. In the same spirit, each contributor should be rendered as an `li`.

File: homework/src/index.js [hunk 11]
```diff
@@ -31,17 +33,96 @@
...
+      select.addEventListener('change', (event) => {
+        div.innerHTML = '';
+
+        container1(data[event.target.value], div);
+
+
+      });
+
+
+    })
```
Comment (remarcmij) [inline homework/src/index.js]: There are a lot of unneeded (and as far as I'm concerned, unwanted) blank lines here.

File: homework/src/App.js (hunk 1 above)
Comment (Samtabanja) [reply homework/src/App.js]: i'm not sure if i can commit here ! but thanks for your feedback ,  and i've just finished updating my homework  for week 2 and 3 , i'm late but i'm doing my best thanks alot .

File: homework/src/App.js [hunk 12]
```diff
@@ -17,12 +17,29 @@ class App {
...
+
+
+      repos.forEach((repo, index) => {
+        Util.createAndAppend('option', select, { 'html': repo.name, 'value': index });
+        this.fetchContributorsAndRender(0);
+        select.addEventListener('change', (e) => {
+          const index = e.target.value;
+          this.fetchContributorsAndRender(index);
+        });
+      });
```
Comment (remarcmij) [inline homework/src/App.js]: I may have overlooked this in my previous review, but lines 37-41 should not be in the `forEach` loop but come after it.

Comment (remarcmij) [reply homework/src/App.js]: One other thing I noticed: your app is not responsive:

samtabanja

Comment (Samtabanja) [reply homework/src/App.js]: thanks dear , ill work on it  and ill make it responsive asap.

**Pull request closed without merging**